WebDriverAgentLib/FBApplicationProcessProxy.m

```diff
@@ -20,11 +20,18 @@ @implementation FBApplicationProcessProxy
 + (instancetype)proxyWithApplicationProcess:(XCUIApplicationProcess *)applicationProcess
 {
   NSParameterAssert(applicationProcess);
-  FBApplicationProcessProxy *proxy = [self.class new];
+  NSParameterAssert([[applicationProcess class] isEqual:XCUIApplicationProcess.class]);
+  FBApplicationProcessProxy *proxy = [[self.class alloc] init];
   proxy.applicationProcess = applicationProcess;
   return proxy;
 }
 
+- (instancetype)init {
+  return self;
```

Comment:
What is the point of init here?

Reply:
NSProxy does not implement ```init``` method.

Reply:
Without implementing ```init``` method, there is an error trying to instantiate ```FBApplicationProxy``` object.

```
failed: caught "NSInvalidArgumentException", "*** +[FBApplicationProcessProxy init]: unrecognized selector sent to class
```

The same for the ```new``` method. Both methods ```init``` and ```new``` are implemented in ```NSOjbect``` and not in ```NSProxy```